crates/ruff_python_ast/src/comparable.rs

```diff
@@ -600,25 +602,115 @@ pub struct ComparableFString<'a> {
     elements: Vec<ComparableFStringElement<'a>>,
 }
 
-impl<'a> From<&'a ast::FString> for ComparableFString<'a> {
-    fn from(fstring: &'a ast::FString) -> Self {
-        Self {
-            elements: fstring.elements.iter().map(Into::into).collect(),
+impl<'a> From<&'a ast::FStringValue> for ComparableFString<'a> {
+    // The approach below is somewhat complicated, so it may
+    // require some justification.
+    //
+    // Suppose given an f-string of the form
+    // `f"{foo!r} one" " and two " f" and three {bar!s}"`
+    // This decomposes as:
+    // - An `FStringPart::FString`, `f"{foo!r} one"` with elements
+    //      - `FStringElement::Expression` encoding `{foo!r}`
+    //      - `FStringElement::Literal` encoding " one"
+    // - An `FStringPart::Literal` capturing `" and two "`
+    // - An `FStringPart::FString`, `f" and three {bar!s}"` with elements
+    //      - `FStringElement::Literal` encoding " and three "
+    //      - `FStringElement::Expression` encoding `{bar!s}`
+    //
+    // We would like to extract from this a vector of (comparable) f-string
+    // _elements_ which alternate between expression elements and literal
+    // elements. In order to do so, we need to concatenate adjacent string
+    // literals. String literals may be separated for two reasons: either
+    // they appear in adjacent string literal parts, or else a string literal
+    // part is adjacent to a string literal _element_ inside of an f-string part.
+    //
+    // So the idea of the code below is as follows:
+    // - Iterate over parts.
+    // - When a literal is encountered, push it to a stack.
```

Comment:
The word `stack` here is a bit confusing because the implementation doesn't make use of a stack. It uses a single "slot" storing the last string literal. 

I think I'm good with the above comment because the most important informatin that I wasn't aware of is that adjacent string literals need to be concatenated.

Reply:
Reworded a bit.